backend/sublet/serializers.py

```diff
@@ -61,6 +61,35 @@ class SubletSerializer(serializers.ModelSerializer):
         many=True, queryset=Amenity.objects.all(), required=False
     )
 
+    def validate_not_draft(self, validated_data, instance=None):
+        if "is_draft" in validated_data and not validated_data["is_draft"]:
+            # check that certain fields are there
+            fields = [
+                "title",
+                "address",
+                "price",
+                "negotiable",
+                "start_date",
+                "end_date",
+                "expires_at",
+            ]
+            if bad_fields := [
+                field
+                for field in fields
+                if (field in validated_data and not validated_data[field])
+                or (
+                    field not in validated_data
+                    and (
+                        instance is None
+                        or not hasattr(instance, field)
+                        or not getattr(instance, field)
+                    )
+                )
+            ]:
+                raise serializers.ValidationError(
```

Comment:
Haha this doesn't work I don't know why haha 😭 will fix

Reply:
thanks haha. is there any way we can break this down? i get that its the least amount of lines but i feel like its getting a little hard to read

Reply:
would really suggest breaking it into first checking if the function came from a PUT/PATCH or a POST

Reply:
so it would seem that patching with "" as the address passes the old address into validated_data??? really not sure why but will investigate further tomorrow.

Reply:
that is weird ...